universal-application-tool-0.0.1/app/views/dev/AwsViewStorageStrategy.java

```diff
@@ -0,0 +1,58 @@
+package views.dev;
+
+import static j2html.TagCreator.form;
+import static j2html.TagCreator.input;
+import static j2html.TagCreator.text;
+import static j2html.attributes.Attr.ENCTYPE;
+
+import j2html.TagCreator;
+import j2html.tags.ContainerTag;
+import services.cloud.StorageUploadRequest;
+import services.cloud.aws.SignedS3UploadRequest;
+import views.HtmlBundle;
+import views.ViewUtils;
+
+/** Strategy class for creating a file upload form for AWS. */
+public class AwsViewStorageStrategy implements FileUploadViewStorageStrategy {
+
+  @Override
+  public ContainerTag getFileUploadForm(
+      ViewUtils viewUtils, StorageUploadRequest storageUploadRequest, HtmlBundle bundle) {
+    if (!(storageUploadRequest instanceof SignedS3UploadRequest)) {
+      return null;
+    }
```

Comment:
suggestion (blocking): avoid returning `null` or assigning `null` to variables. If a value may not be present change the return type to `Optional` and return an `Optiona.empty()`

question (blocking): Is hitting this conditional an expected part of user flow or is it guarding an error case? if the latter please log an error before redirecting. if the former please add a comment.

Reply:
It is an error case so I have updated to throw an exception. It should technically never happen though